azurelinuxagent/ga/monitor.py

```diff
@@ -36,58 +36,148 @@
+            msg = "hostplugin:{0};protocol:{1};other:{2}".format(hostplugin_errors, protocol_errors, other_errors)
+            add_event(op=WALAEventOperation.HttpErrors, message=msg)
+
+
+class ReportNetworkConfigurationChangesOperation(PeriodicOperation):
+    """
+    Periodic operation to check and log changes in network configuration.
+    """
+
+    def __init__(self):
+        super(ReportNetworkConfigurationChangesOperation, self).__init__(
+            name="report network configuration changes",
+            operation=self._operation_impl,
+            period=datetime.timedelta(minutes=1))
+        self.osutil = get_osutil()
+        self.last_route_table_hash = b''
+        self.last_nic_state = {}
+
+    def _operation_impl(self):
+        raw_route_list = self.osutil.read_route_table()
+        digest = hash_strings(raw_route_list)
+        if digest != self.last_route_table_hash:
+            self.last_route_table_hash = digest
+            route_list = self.osutil.get_list_of_routes(raw_route_list)
+            logger.info("Route table: [{0}]".format(",".join(map(networkutil.RouteEntry.to_json, route_list))))
+
+        nic_state = self.osutil.get_nic_state()
+        if nic_state != self.last_nic_state:
+            description = "Initial" if self.last_nic_state == {} else "Updated"
+            logger.info("{0} NIC state: [{1}]".format(description, ", ".join(map(str, nic_state.values()))))
+            self.last_nic_state = nic_state
+
+
 class MonitorHandler(object):
     # telemetry
     EVENT_COLLECTION_PERIOD = datetime.timedelta(minutes=1)
-    # host health
-    TELEMETRY_HEARTBEAT_PERIOD = datetime.timedelta(minutes=30)
-    # cgroup data period
-    CGROUP_TELEMETRY_POLLING_PERIOD = datetime.timedelta(minutes=5)
     # host plugin
     HOST_PLUGIN_HEARTBEAT_PERIOD = datetime.timedelta(minutes=1)
     HOST_PLUGIN_HEALTH_PERIOD = datetime.timedelta(minutes=5)
     # imds
     IMDS_HEARTBEAT_PERIOD = datetime.timedelta(minutes=1)
     IMDS_HEALTH_PERIOD = datetime.timedelta(minutes=3)
-    # log network configuration
-    LOG_NETWORK_CONFIGURATION_PERIOD = datetime.timedelta(minutes=1)
-    # Resetting loggers period
-    RESET_LOGGERS_PERIOD = datetime.timedelta(hours=12)
 
     def __init__(self):
         self.osutil = get_osutil()
         self.imds_client = None
 
         self.event_thread = None
         self._periodic_operations = [
-            PeriodicOperation("reset_loggers", self.reset_loggers, self.RESET_LOGGERS_PERIOD),
+            ResetPeriodicLogMessagesOperation(),
             PeriodicOperation("collect_and_send_events", self.collect_and_send_events, self.EVENT_COLLECTION_PERIOD),
-            PeriodicOperation("send_telemetry_heartbeat", self.send_telemetry_heartbeat, self.TELEMETRY_HEARTBEAT_PERIOD),
-            PeriodicOperation("poll_telemetry_metrics usage", self.poll_telemetry_metrics, self.CGROUP_TELEMETRY_POLLING_PERIOD),
+            ReportNetworkErrorsOperation(),
```

Comment:
I left these 3 operations in the monitor class because I still need to unravel their dependency on the protocol